cli/src/main/java/com/box/l10n/mojito/cli/command/DateTimeConverter.java

```diff
@@ -1,11 +1,36 @@
 package com.box.l10n.mojito.cli.command;
 
 import com.beust.jcommander.IStringConverter;
-import org.joda.time.DateTime;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeParseException;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DateTimeConverter implements IStringConverter<ZonedDateTime> {
+
+  /** logger */
+  static Logger logger = LoggerFactory.getLogger(DateTimeConverter.class);
 
-public class DateTimeConverter implements IStringConverter<DateTime> {
   @Override
-  public DateTime convert(String dateAsText) {
-    return DateTime.parse(dateAsText);
+  public ZonedDateTime convert(String dateAsText) {
+    ZonedDateTime converted = null;
+
+    if (StringUtils.isNotEmpty(dateAsText)) {
+      try {
+        Instant instant = Instant.ofEpochMilli(Long.parseLong(dateAsText));
+        converted = ZonedDateTime.ofInstant(instant, ZoneId.systemDefault());
+      } catch (NumberFormatException nfe) {
+        try {
+          converted = ZonedDateTime.parse(dateAsText);
+        } catch (DateTimeParseException e) {
+          logger.warn("Unable to parse the date: " + dateAsText);
```

Comment:
The only place I see that uses this converter is in the CommitCreateCommand which is parsing the date from the git log in ISO 8601 format e.g. 2021-09-28T15:30:00+01:00 so I'm not sure if we need the Long parsing logic.

If we do decide to keep it we should probably throw an exception in the event that we are unable to parse the date to ensure the cli throws an error to the caller and doesn't create an invalid commit in the Mojito DB.

Reply:
Thanks Mark. I decided to add the Long parsing logic here because the other DateTime converter in WebApp has it, and I think would makes sense to have it here as well.
https://github.com/pinterest/mojito/blob/master/webapp/src/main/java/com/box/l10n/mojito/rest/textunit/StringToDateTimeConverter.java#L26

About throwing an exception instead of logging it, you are right, I shouldn't change the behavior if parsing errors occur.